### DISCUSSION TITLE
Artist Profile - Configurable "TOP SONGS"

### DISCUSSION THREAD

**Discussion by mist2t** [navidrome/navidrome#3585] [Ideas]
If the Artist Profile page redesign will implement a TOP SONGS section (related PR: #1540 ) it will be cool if this section can be configurable in settings (or a future admin area / dashboard)

The original PR (#1540 ) uses Last.FM for top songs data but allowing some config options for "Top Songs" will be amazing. 

This will allow users to create a custom "best of" list of all songs from every artist, without having to create individual playlists for each artist (cluttering the "normal" playlists).

Configuration Options:

1. **Top Songs by Ratings** (pull all 5 star rated songs from artist discography / albums *optional: configurable rating threshold)
2. **Top Songs by Faves** (pull all songs favorited / "loved" by user from artist discography / albums)
3. **Top Songs by Plays** (pull top songs by number of plays, most played first)
4. **Top Songs by Social Sites** (pull top songs referenced from external sources, ex Last FM)
5. **All songs** (pull all songs from the artist)

* any other ideas for criteria

Of course, this "Top Songs" section should also have a play / shuffle / queue option.

**Usage scenario**: If I want to listen to my "picks" (faves / 5 star songs) from any particular artist discography I can just search for the artist and play the "Top Songs" for that artist. Without having to create gazillion of "best of" playlists for gazillion of artists :)

Edited: 
- Simplified the post :)
- Added more suggested criteria

**Comment by MRodneyMcKay** [comment]
Or, as an other option, it would be cool, if navidrome could build this list based on our plays. So, essentially, this would be a most played list by the given artist.

**Reply by mist2t** [reply]
Yes, totally. This could be a super nice additional criteria : Top Songs by number of plays (most played first)
Basically this configurable "Top Songs" section could allow for curating a list of songs on  artist profile, by various criteria.

I've been missing this so badly in other software.

**Comment by candideu** [comment]
Related (ish): https://github.com/navidrome/navidrome/issues/961

Feishin has it, but I would love to see it in the Navidrome web UI too:

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/62ffe5f2-8296-4fb0-9dec-967cc6394186" />
